packages/react-imready/src/react-imready/useImReady.tsx

```diff
@@ -26,106 +28,23 @@ import { ImReadyProps, ImReadyValue } from "./types";
-    const [readyCount, setReadyCount] = useState(0);
-    const [errorCount, setErrorCount] = useState(0);
-    const [totalErrorCount, setTotalErrorCount] = useState(0);
-    const [totalCount, setTotalCount] = useState(0);
-    const [isPreReady, setIsPreReady] = useState(false);
-    const [isReady, setIsReady] = useState(false);
-    const [hasError, setHasError] = useState(false);
-    const [isPreReadyOver, setIsPreReadyOver] = useState(false);
-
-    function register<T extends HTMLElement>(ref?: Ref<T>) {
-        return (instance: T | null) => {
-            if (instance) {
-                children.push(instance);
-            }
-            if (!ref) {
-                return;
-            }
-            const refType = typeof ref;
-
-            if (refType === "function") {
-                (ref as RefCallback<T>)(instance);
-            } else {
-                (ref as MutableRefObject<T | null>).current = instance;
-            }
-        };
-    }
-
-    useEffect(() => {
-        const im = new ImReady(options);
-
-        let checkedElements = children;
-        if (selector) {
-            checkedElements = checkedElements.reduce((prev, cur) => {
-                return [...prev, ...cur.querySelectorAll<HTMLElement>(selector)];
-            }, [] as HTMLElement[]);
-;        }
-        im.check(checkedElements).on("error", e => {
-            if (useError) {
-                setHasError(true);
-                setErrorCount(e.errorCount);
-                setTotalErrorCount(e.totalErrorCount);
-            }
-        }).on("preReadyElement", e => {
-            if (usePreReadyElement) {
-                setPreReadyCount(e.preReadyCount);
-            }
-        }).on("readyElement", e => {
-            if (useReadyElement) {
-                setReadyCount(e.readyCount);
-                setIsPreReadyOver(e.isPreReadyOver);
-            }
-        }).on("preReady", () => {
-            if (usePreReady) {
-                setIsPreReady(true);
-            }
-        }).on("ready", () => {
-            if (useReady) {
-                setIsReady(true);
-            }
-        });
-        setPreReadyCount(0);
-        setReadyCount(0);
-        setTotalErrorCount(0);
-        setErrorCount(0);
-        setTotalCount(im.getTotalCount());
-        setIsReady(false);
-        setIsPreReady(false);
-        setHasError(false);
-        setIsPreReadyOver(false);
-
-        return () => {
-            im.destroy();
-        };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
-
-    return {
-        preReadyCount,
-        readyCount,
-        totalCount,
-        errorCount,
-        totalErrorCount,
-        isPreReady,
-        isReady,
-        hasError,
-        isPreReadyOver,
-        register,
-    };
+export function useImReady(
+  props: Partial<ImReadyHooksProps>
+): ReactImReadyResult {
+  return useReactive({
+    data() {
+      return {
+        props: {
+          usePreReady: true,
```

Comment:
internally
  You do not need to use usePreReady, usePreReadyElement, useReady, useReadyElement, or useError.

usImReady = usePreReadyElement = useReady = useReadyElement